src/query/expression/src/evaluator.rs

```diff
@@ -951,13 +951,96 @@ impl<'a> Evaluator<'a> {
         unreachable!("expr is not a set returning function: {expr}")
     }
 
+    fn run_array_fold(&self, column: &Column, expr: &Expr) -> Result<Value<AnyType>> {
+        if column.data_type().is_null() || column.len() < 1 {
+            return Ok(Value::Scalar(Scalar::Null));
+        }
+
+        let first_element = column.index(0).unwrap();
+        let cast_value = self.run_cast(
+            None,
+            &first_element.infer_data_type(),
+            expr.data_type(),
+            Value::Scalar(first_element.to_owned()),
+            None,
+        )?;
+        let mut arg0 = cast_value.as_scalar().unwrap().as_ref();
+        let mut result = cast_value.to_owned();
+
+        for i in 1..column.len() {
+            let arg1 = column.index(i).unwrap();
+            let entries = if expr.data_type().is_nullable() {
+                vec![
+                    BlockEntry::new(
+                        if arg0.is_null() {
+                            DataType::Null
```

Comment:
the data type can get by `column.data_type()`, don't need to check by `arg0` every time.